backend/src/main/java/com/capstone/moneytree/controller/StockController.java

```diff
@@ -0,0 +1,83 @@
+package com.capstone.moneytree.controller;
+
+import com.capstone.moneytree.service.api.StockMarketDataService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import com.capstone.moneytree.service.api.StockService;
+import pl.zankowski.iextrading4j.api.stocks.*;
+import pl.zankowski.iextrading4j.api.stocks.v1.BatchStocks;
+import pl.zankowski.iextrading4j.api.stocks.v1.KeyStats;
+import pl.zankowski.iextrading4j.api.stocks.v1.News;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Size;
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/stockmarket")
+public class StockController extends ApiController {
+
+   @Autowired
+   private StockMarketDataService stockMarketDataService;
+   private static final Logger LOG = LoggerFactory.getLogger(StockController.class);
+
+   /**
+    * Obtain json object containing company quote, info, news, keystats and logo
+    *
+    * @param symbol
+    * @return batchStocks
+    */
+   @GetMapping("/batch/{symbol}")
+   public ResponseEntity<BatchStocks> getBatchStocksBySymbol(@Valid @PathVariable(name = "symbol") String symbol) {
+      BatchStocks batchStocks = stockMarketDataService.getBatchStocksBySymbol(symbol);
+      return ResponseEntity.ok(batchStocks);
```

Comment:
We need to validate that the batchStocks response is not null.

Reply:
Same for all endpoints

Reply:
Absolutely, we can make a @ControllerAdvice class which allows you to handle exceptions across the whole application. It listens for errors to be thrown from the controller methods and enables you to handled them all in one single file.